### ISSUE TITLE
Fetch region information dynamically

### ISSUE THREAD

**Issue by steffenkleinle** [digitalfabrik/entitlementcard#538]
**Is your feature request related to a problem? Please describe.**
The regions are hardcoded in `EakBayernRegions.kt` instead of dynamically extracting them from the correct dynamic source (probably app-daten.xml).

**Describe the solution you'd like**
- Regions should be collected from the data import (in case of bayern the app-daten.xml)
- Region that do not contain a name (freinet), the name can be resolved from `freinetagencies` 
- Regions that do not contain a `regionIdentifier` can be mapped via postal code from `plz_ort_bayern.csv`

**Describe alternatives you've considered**
None

**Additional context**
Data source: https://www.ehrenamt.bayern.de/xml-json/app-daten.xml

Blocked: The freinet regions do not contain proper naming for the region that we can use (check freinetagencies table).
The region name is also used for the application form (Step 1) so we should have correct prefix and name for all regions

**Comment by maxammann** [comment]
I updated the first bullet point. There is probably no external data source additionally to the app-daten.xml. There is only the app-daten.xml

**Comment by klinzo** [comment]
> * I updated the first bullet point. There is probably no external data source additionally to the app-daten.xml. There is only the app-daten.xml
> 
>     * I added the second bullet point.

However there is no kreisschlüssel in the app daten xml, how can we map that to regions? Also in theory it may happen that a landkreis or stadt does not publish stores but has card holders. Since cards only belong to a region, this may lead to problems as the region may not exist anymore.

**Comment by maxammann** [comment]
> > * I updated the first bullet point. There is probably no external data source additionally to the app-daten.xml. There is only the app-daten.xml
> >   
> >   * I added the second bullet point.
> 
> However there is no kreisschlüssel in the app daten xml, how can we map that to regions? Without a kreisschlüssel we need to change the database scheme again since it has to be nullable then. Also in theory it may happen that a landkreis or stadt does not publish stores but has card holders. Since cards only belong to a region, this may lead to problems as the region may not exist anymore.

True, so the kreischlüssel should be optional. It should also be possible that Users overwrite the imported kreischlüssel if its imported (future work).

So regions should not be deleted. As long as there are stores or cards in a region it will exist.

**Comment by f1sh1918** [comment]
The regions have to be mapped to the stores via `normale` agency ID. Check freinet documentation for further information

**Comment by f1sh1918** [comment]
blocked by #1762

**Comment by f1sh1918** [comment]
@steffenkleinle 
i think we should split this issue and discuss every bulletpoint.
The most important thing is to map stores to a region (bulletpoint 3), this should be done first when #1762 is ready

To bullet point one:
I think there is a second data source that is not from freinet and has no `agencyId` and therefore we don't get region information we need. There is a region information endpoint from freinet but they only have regions that use freinet.

To bullet point two:
I think thats a nice feature but not really connected to the mapping here
You could create a separate feature if this functionality is needed in the future. No idea which purpose this should have now

**Comment by f1sh1918** [comment]
Map stores to region moved to  #1968

**Comment by f1sh1918** [comment]
I really don't think that the lbe and freinet xml source is stable enough to base our region creation on that @connium
I would prefer to add a section for project admins where they can add new regions (project depending l)

**Comment by connium** [comment]
There aren't many regions missing at the moment and it's not clear if they will join at all. So I think our current solution is sufficient and if a new region wants to join, this can be fixed manually (no need to handle unstable data sources 😉 ).
